core/src/main/scala/org/bitcoins/core/protocol/dlc/ContractInfo.scala

```diff
@@ -127,167 +263,149 @@ case class ContractInfo(
+          case DLCAccept.NegotiationFieldsV1(acceptRoundingIntervals) =>
+            descriptor.roundingIntervals.minRoundingWith(
+              acceptRoundingIntervals)
+          case DLCAccept.NoNegotiationFields => descriptor.roundingIntervals
+          case _: DLCAccept.NegotiationFieldsV2 =>
+            throw new IllegalArgumentException(
+              s"Cannot use disjoint union negotiation fields for a SingleContractInfo, $negotiationFields")
+        }
 
+        if (
+          newTotalCollateral == totalCollateral && newRoundingIntervals == descriptor.roundingIntervals
+        ) {
+          this
+        } else {
           val newDescriptor =
             descriptor.copy(roundingIntervals = newRoundingIntervals)
           val contractOraclePair =
             ContractOraclePair.NumericPair(newDescriptor, oracleInfo)
-          ContractInfo(totalCollateral = newTotalCollateral,
-                       contractOraclePair = contractOraclePair)
-      }
+          SingleContractInfo(totalCollateral = newTotalCollateral,
+                             contractOraclePair = contractOraclePair)
+        }
     }
   }
 }
 
-object ContractInfo
-    extends TLVDeserializable[ContractInfoV0TLV, ContractInfo](
+object SingleContractInfo
+    extends TLVDeserializable[ContractInfoV0TLV, SingleContractInfo](
       ContractInfoV0TLV) {
 
   lazy val dummy: ContractInfo = fromTLV(ContractInfoV0TLV.dummy)
 
-  override def fromTLV(tlv: ContractInfoV0TLV): ContractInfo = {
+  override def fromTLV(tlv: ContractInfoV0TLV): SingleContractInfo = {
     val contract = ContractDescriptor.fromTLV(tlv.contractDescriptor)
     val oracleInfo = OracleInfo.fromTLV(tlv.oracleInfo)
     val contractOraclePair =
       ContractOraclePair.fromDescriptorOracle(contract, oracleInfo)
-    ContractInfo(tlv.totalCollateral, contractOraclePair)
+    SingleContractInfo(tlv.totalCollateral, contractOraclePair)
   }
 
   def apply(
       enumDescriptor: EnumContractDescriptor,
-      enumOracleInfo: EnumOracleInfo): ContractInfo = {
+      enumOracleInfo: EnumOracleInfo): SingleContractInfo = {
     val enumPair = ContractOraclePair.EnumPair(enumDescriptor, enumOracleInfo)
-    ContractInfo(totalCollateral = enumDescriptor.values.maxBy(_.toLong),
-                 enumPair)
+    SingleContractInfo(totalCollateral = enumDescriptor.values.maxBy(_.toLong),
+                       enumPair)
   }
 
   def apply(
       totalCollateral: Satoshis,
       contractDescriptor: ContractDescriptor,
-      oracleInfo: OracleInfo): ContractInfo = {
-    ContractInfo(
+      oracleInfo: OracleInfo): SingleContractInfo = {
+    SingleContractInfo(
       totalCollateral,
       ContractOraclePair.fromDescriptorOracle(contractDescriptor, oracleInfo))
   }
 }
+
+case class DisjointUnionContractInfo(contracts: Vector[SingleContractInfo])
+    extends ContractInfo
+    with TLVSerializable[ContractInfoV1TLV] {
+
+  override val totalCollateral: Satoshis = contracts.head.totalCollateral
```

Comment:
nit: should add a `require(contracts.nonEmpty)` so a better error message is given